proto/stream_plan.proto

```diff
@@ -423,12 +423,10 @@ message LookupNode {
 
 // WatermarkFilter needs to filter the upstream data by the water mark.
 message WatermarkFilterNode {
-  // The expression to calculate the watermark value.
-  expr.ExprNode watermark_expr = 1;
-  // The column the event time belongs.
-  uint64 event_time_col_idx = 2;
-  // The table used to persist watermark, the key is vnode.
-  catalog.Table table = 3;
+  // The watermark descs
+  repeated catalog.WatermarkDesc watermark_descs = 1;
```

Comment:
Don't worry. `repeated` field is compatible with non-repeated one, so you may leave them as non-repeated now and change it until needed, in order to make the code cleaner.